elements/radio-group/radio-group.js

```diff
@@ -1,151 +1,233 @@
-
-// For interop reasons[1], we’ll have to maintain our map
-// for semantic key names.
-const KEYCODE = {
-  DOWN: 40,
-  LEFT: 37,
-  RIGHT: 39,
-  SPACE: 32,
-  UP: 38
-};
-
-// The `RadioGroup` does all the heavy lifting. It attaches
-// event listeners, adjust ARIA attribute values and provides
-// the API for this element.
-class RadioGroup extends HTMLElement {
-  constructor() {
-    super();
+    /**
+     * The `RadioButton` sets its initial ARIA attributes when it's attached to
+     * the DOM. The surrounding RadioGroup handles dynamic changes to the ARIA
+     * attributes. The RadioButton should always set a `role` of `radio`, and
+     * should check to see if its `tabindex` and `aria-checked` values have been
+     * set by the user. Otherwise, it should use default values.
+     */
+    connectedCallback() {
+      this.setAttribute('role', 'radio');
+      this.setAttribute('tabindex', this.getAttribute('tabindex') || -1);
+      this.setAttribute('aria-checked', this.getAttribute('aria-checked') || false);
+    }
   }
 
-  connectedCallback() {
-    this.addEventListener('keydown', this.onKeyDown);
-    this.addEventListener('click', this._onClick);
+  /**
+   * Define a custom element, `<radio-button>`, and associate it with the
+   * `RadioButton` class.
+   */
+  window.customElements.define('radio-button', RadioButton);
+
+  /**
+   * `RadioGroup` is responsible for handling user input, and updating the state
+   * of its `RadioButton` children. It uses the [roving tabindex](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets#Technique_1_Roving_tabindex)
+   * technique to respond to keyboard events. This ensures that the entire
+   * `RadioGroup` is a single tabstop, and tabbing into the `RadioGroup` will
+   * always focus the previously checked item, if one exists.
+   */
+  class RadioGroup extends HTMLElement {
+    constructor() {
+      super();
+    }
 
-    this.setAttribute('role', 'radiogroup');
-    // Quick and dirty set first child to tabindex 0.
-    // Really we should check to see if anyone already has tabindex 0,
-    // but we don’t care. Lol.
-    this.querySelector('[role="radio"]').setAttribute('tabindex', 0);
+    /**
+     * The `RadioGroup` sets its ARIA role to `radiogroup` and sets the
+     * `tabindex` on its first `RadioButton` child to 0 if no other child is
+     * already set to `tabindex=0`. This makes the first `RadioButton` focusable.
+     * This assumes `RadioButton` children are already in the DOM and their
+     * definition has already been loaded. For a more robust implementation you
+     * might consider using a Mutation Observer to detect if children are
+     * present yet. The `RadioGroup also adds listeners for keyboard and mouse
+     * events. 
+     */
+    connectedCallback() {
+      this.setAttribute('role', 'radiogroup');
+      let hasCheckedChild = this.querySelector('[role="radio"][tabindex="0"]');
```

Comment:
Good question. The thought has occurred to me but I chose to ignore it for simplicity. I just did a demo to see what the native radio group does and it looks like if more than one child is checked, it styles whichever one is latest in the DOM. But it leaves `checked` on all of them.

Is there an approach you would prefer?